### PR TITLE
Change CSV annotations to `images.<version>.<imageName>`

### PR DESCRIPTION
**Pull request by dgn**
That's where the operator is looking for them.

### CODE REVIEW

**Review by FilipB — APPROVED** [review]

**Comment by bmangoen** [comment]
I've tested it and the sail-operator pod is in the `CrashLoopBackOff` state with this following error:
```
$ kubectl -nsail-operator logs deploy/sail-operator 
2024-08-16T09:21:52Z    INFO    setup   version.BuildInfo{Version:"0.1.0", GitRevision:"27acb19e72427ae02e9911f585c6886a3dde29b4", BuildStatus:"Modified", GitTag:"unknown", GoVersion:"go1.22.5", GoArch:"linux/amd64", ControllerRuntime:"v0.18.3"}
2024-08-16T09:21:52Z    INFO    setup   reading config
2024-08-16T09:21:52Z    ERROR   setup   unable to read config file at /etc/sail-operator/config.properties      {"error": "missing required key images.latest.istiod"}
main.main
        ./main.go:89
runtime.main
        runtime/proc.go:271
```

**Comment by dgn** [comment]
> I've tested it and the sail-operator pod is in the `CrashLoopBackOff` state with this following error:
> 
> ```
> $ kubectl -nsail-operator logs deploy/sail-operator 
> 2024-08-16T09:21:52Z    INFO    setup   version.BuildInfo{Version:"0.1.0", GitRevision:"27acb19e72427ae02e9911f585c6886a3dde29b4", BuildStatus:"Modified", GitTag:"unknown", GoVersion:"go1.22.5", GoArch:"linux/amd64", ControllerRuntime:"v0.18.3"}
> 2024-08-16T09:21:52Z    INFO    setup   reading config
> 2024-08-16T09:21:52Z    ERROR   setup   unable to read config file at /etc/sail-operator/config.properties      {"error": "missing required key images.latest.istiod"}
> main.main
>         ./main.go:89
> runtime.main
>         runtime/proc.go:271
> ```

Thanks! Checking...

**Comment by dgn** [comment]
Fixed! Thank you very much @bmangoen

File: hack/patch-csv.sh [hunk 1]
```diff
@@ -88,14 +91,14 @@ versions="$( ${YQ} '.versions[].name' "${VERSIONS_YAML_FILE}" )"
 
 for version in ${versions}; do
   version_underscore=${version//./_}
-  for component_name in "${!COMPONENTS[@]}"; do    
+  for component_name in "${!COMPONENTS[@]}"; do
     name="${version_underscore}.${COMPONENTS[$component_name]}"
```
Comment (bmangoen) [inline hack/patch-csv.sh:95]: Would it be easier to read/understand if you change this to `name="${version_underscore}.${$component_name}"` and revert the other changes?

Comment (dgn) [reply hack/patch-csv.sh]: makes sense. made the change

**Pull request merged**